macros/get_optional_fields.sql

```diff
@@ -0,0 +1,286 @@
+{% macro get_iab_context_fields(table_prefix = none) %}
+    {{ return(adapter.dispatch('get_iab_context_fields', 'snowplow_web')(table_prefix)) }}
+{%- endmacro -%}
+
+{% macro get_ua_context_fields(table_prefix = none) %}
+    {{ return(adapter.dispatch('get_ua_context_fields', 'snowplow_web')(table_prefix)) }}
+{%- endmacro -%}
+
+{% macro get_yauaa_context_fields(table_prefix = none) %}
+    {{ return(adapter.dispatch('get_yauaa_context_fields', 'snowplow_web')(table_prefix)) }}
+{%- endmacro -%}
+
+{# iab fields #}
+{% macro default__get_iab_context_fields(table_prefix = none) %}
+    {%- if var('snowplow__enable_iab', false) -%}
+        {% if table_prefix %}table_prefix.{% endif %}category,
+        {% if table_prefix %}table_prefix.{% endif %}primary_impact,
+        {% if table_prefix %}table_prefix.{% endif %}reason,
+        {% if table_prefix %}table_prefix.{% endif %}spider_or_robot
+    {%- else -%}
+        cast(null as {{ type_string() }}) as category,
+        cast(null as {{ type_string() }}) as primary_impact,
+        cast(null as {{ type_string() }}) as reason,
+        cast(null as boolean) as spider_or_robot
+    {%- endif -%}
+{% endmacro %}
+
+{% macro databricks__get_iab_context_fields(table_prefix = none) %}
```

Comment:
You should probably also add a `spark__` version of these which just calls the Databricks one

Reply:
I've just posted in the dbt slack because I was under the impression that they have inheritence in once direction, but I don't _know_ that that's true, and I can't tell which way it would be (their docs only specify it for postgres/redshift, but I feel like they should be... https://docs.getdbt.com/reference/dbt-jinja-functions/dispatch#adapter-inheritance)

Reply:
AFAIK it's Databricks inheriting from Spark, *not* the other way around, but good to double check, will follow the convo there!

Reply:
Cool, will fix in another branch and also check other macros at the same time